### PR TITLE
`poetry.lock` added

### PR DESCRIPTION
**Pull request by Raalsky**
Moving dev dependencies lower the time needed for `poetry lock` command. `poetry.lock` file should allow Dependantbot checks.

---

## Before submitting checklist

- [ ] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ style = "semver"
 pattern = "default-unprefixed"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = ">=3.7,<=3.12"
```
Comment (szysad) [inline pyproject.toml:12]: Does it mean that we can now develop in python 3.10?? 🤩

File: .github/actions/install-package/action.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ runs:
 
       - name: Install dependencies
         run: |
-          pip install ".[dev,e2e]"
+          pip install -r dev_requirements.txt
```
Comment (szysad) [inline .github/actions/install-package/action.yml:33]: note: when merged we should update [notion env setup for client](https://www.notion.so/neptuneai/PyCharm-for-neptune-client-f8c8ba96d1c6469fa833f74dc302b2b7?pvs=4#6ff75edc9ed74b7e8d0aface7d20f612)

**Comment by szysad** [comment]
What is the need for this change? Seems like a big one and not sure why are we doing it.

**Comment by szysad** [comment]
⌛ [E2E](https://github.com/neptune-ai/neptune-client/actions/runs/7869242562)

File: pyproject.toml (hunk 1 above)
Comment (Raalsky) [reply pyproject.toml:12]: As long as the code itself is still compatible with Python 3.7 -> Sure 😈

File: .github/actions/install-package/action.yml (hunk 2 above)
Comment (Raalsky) [reply .github/actions/install-package/action.yml:33]: Yes, it's noted on my TODO :wink:

**Review by szysad — APPROVED** [review]

**Pull request merged**